src/FilenameParsing/FileIDHelperResolutionStrategy.php

```diff
@@ -435,28 +435,33 @@ public function findVariants($tuple, Filesystem $filesystem)
 
             // Find the correct folder to search for possible variants in
             $folder = $helper->lookForVariantIn($parsedFileID);
-            $possibleVariants = $filesystem->listContents($folder, $helper->lookForVariantRecursive())->toArray();
+            $possibleVariantsGenerator = $filesystem->listContents($folder, $helper->lookForVariantRecursive());
 
-            // Flysystem returns array of meta data abouch each file, we remove directories and map it down to the path
-            $possibleVariants = array_filter($possibleVariants ?? [], function ($possibleVariant) {
-                return $possibleVariant['type'] !== 'dir';
+            // Flysystem returns generator of meta data abouch each file, we remove directories and map it down to the path
+            $possibleVariantPaths = $possibleVariantsGenerator->filter(function ($item) {
+                return !$item->isDir();
+            })->map(function ($item) {
+                return $item->path();
             });
```

Comment:
```suggestion
            $possibleVariantPaths = $possibleVariantsGenerator
                ->filter(fn($item) => !$item->isDir())
                ->map(fn($item) => $item->path());
```
Easier to read with arrow functions

Reply:
Good call. Done.